server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainMappingsRoutes.java

```diff
@@ -0,0 +1,155 @@
+
+    private final RecipientRewriteTable recipientRewriteTable;
+    private final JsonTransformer jsonTransformer;
+
+    @Inject
+    public DomainMappingsRoutes(final RecipientRewriteTable recipientRewriteTable, final JsonTransformer jsonTransformer) {
+        this.recipientRewriteTable = recipientRewriteTable;
+        this.jsonTransformer = jsonTransformer;
+    }
+
+    @Override
+    public void define(final Service service) {
+        service.get(DOMAIN_MAPPINGS, this::get, jsonTransformer);
+        service.put(SPECIFIC_MAPPING, this::addDomainMapping);
+        service.delete(SPECIFIC_MAPPING, this::removeDomainMapping);
+    }
+
+    @PUT
+    @Path(SPECIFIC_MAPPING_PATH)
+    @ApiOperation(value = "Creating domain mapping between source and destination domains.")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "string", name = FROM_DOMAIN, paramType = "path")
+    })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "Ok"),
+            @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Domain name is invalid"),
+            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+                    message = "Internal server error - Something went bad on the server side.")
+    })
+    public HaltException addDomainMapping(Request request, Response response) {
+        doMapping(request, recipientRewriteTable::addAliasDomainMapping);
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    @DELETE
+    @Path(SPECIFIC_MAPPING_PATH)
+    @ApiOperation(value = "Removes domain mapping between source and destination domains.")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "string", name = FROM_DOMAIN, paramType = "path")
+    })
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "Ok"),
+            @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Domain name is invalid"),
+            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+                    message = "Internal server error - Something went bad on the server side.")
+    })
+    public HaltException removeDomainMapping(Request request, Response response) {
+        doMapping(request, recipientRewriteTable::removeAliasDomainMapping);
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    @GET
+    @Path(DOMAIN_MAPPINGS)
+    @ApiOperation(value = "Lists all domain mappings.")
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.OK_200, message = "Domain mappings.", responseContainer = "Map"),
+            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+                    message = "Internal server error - Something went bad on the server side.")
+    })
+    public Map<String, List<String>> get(Request request, Response response) throws RecipientRewriteTableException {
```

Comment:
It smells like a MultiList. Guavate has some collectors for this. You may need to load Guava's Jackson module in JsonTransformer also...

Reply:
If you mean `MultiMap` then, yes, of course it is!